[DA] ToastView 실패 케이스 구현

kimkyunghun3 opened the pull request:
# 개요
- 🔗  이슈링크 : #120 

## 작업 내용
- ToastView 케이스에 따라 다른 뷰를 보이도록 구현

### 스크린샷
## 케이스에 따라 다른 ToastView

| 성공 | 실패 | 네트워크실패 |
|:---:|:---:|:---:|
|<img src ="https://user-images.githubusercontent.com/52434820/183916062-77c45cf7-3d03-439d-8cf1-0cdff341b622.png" width ="200">|<img src ="https://user-images.githubusercontent.com/52434820/183916055-092f6e16-f0f7-490b-a14b-8fe02c9d1203.png" width ="200">|<img src ="https://user-images.githubusercontent.com/52434820/183916069-76146632-1e6e-4525-ad02-0136a32264c4.png" width ="200">|

kimkyunghun3 committed:
:recycle: ToastView enum 케이스 변경

# :tada: = 프로젝트 시작 (#)
# :sparkles: = create 파일 생성 시 (#)
# :bulb: = update 코드 및 UI 업데이트, 기능 구현 시 (#)
# :rocket: = release 배포 시 (#)
# :recycle: = refactor 리팩토링 시 (기능은 같은데 코드가 달라지는경우) (#)
# :memo: = 문서 작성 시 (#)
# :art: = 디자인 시스템 관련 내용 (Color / Asset) (#)
# :hammer: = 테스트 코드 작성 (#)
################

# 커밋 타입 : 커밋 제목 (계정명/저장소명#이슈번호) 형식으로 작성하며 제목은 최대 50글자 정도로만 입력
# 제목을 아랫줄에 작성, 제목 끝에 마침표 금지, 무엇을 했는지 명확하게 작성

kimkyunghun3 committed:
:recycle: ToastView 필요한 메서드 추가

# :tada: = 프로젝트 시작 (#)
# :sparkles: = create 파일 생성 시 (#)
# :bulb: = update 코드 및 UI 업데이트, 기능 구현 시 (#)
# :rocket: = release 배포 시 (#)
# :recycle: = refactor 리팩토링 시 (기능은 같은데 코드가 달라지는경우) (#)
# :memo: = 문서 작성 시 (#)
# :art: = 디자인 시스템 관련 내용 (Color / Asset) (#)
# :hammer: = 테스트 코드 작성 (#)
################

# 커밋 타입 : 커밋 제목 (계정명/저장소명#이슈번호) 형식으로 작성하며 제목은 최대 50글자 정도로만 입력
# 제목을 아랫줄에 작성, 제목 끝에 마침표 금지, 무엇을 했는지 명확하게 작성

kimkyunghun3 committed:
:recycle: showToastView 로직 변경

# :tada: = 프로젝트 시작 (#)
# :sparkles: = create 파일 생성 시 (#)
# :bulb: = update 코드 및 UI 업데이트, 기능 구현 시 (#)
# :rocket: = release 배포 시 (#)
# :recycle: = refactor 리팩토링 시 (기능은 같은데 코드가 달라지는경우) (#)
# :memo: = 문서 작성 시 (#)
# :art: = 디자인 시스템 관련 내용 (Color / Asset) (#)
# :hammer: = 테스트 코드 작성 (#)
################

# 커밋 타입 : 커밋 제목 (계정명/저장소명#이슈번호) 형식으로 작성하며 제목은 최대 50글자 정도로만 입력
# 제목을 아랫줄에 작성, 제목 끝에 마침표 금지, 무엇을 했는지 명확하게 작성

kimkyunghun3 committed:
:recycle: toastView 성공, 실패, 에러 케이스에 따라 구분 로직 변경

# :tada: = 프로젝트 시작 (#)
# :sparkles: = create 파일 생성 시 (#)
# :bulb: = update 코드 및 UI 업데이트, 기능 구현 시 (#)
# :rocket: = release 배포 시 (#)
# :recycle: = refactor 리팩토링 시 (기능은 같은데 코드가 달라지는경우) (#)
# :memo: = 문서 작성 시 (#)
# :art: = 디자인 시스템 관련 내용 (Color / Asset) (#)
# :hammer: = 테스트 코드 작성 (#)
################

# 커밋 타입 : 커밋 제목 (계정명/저장소명#이슈번호) 형식으로 작성하며 제목은 최대 50글자 정도로만 입력
# 제목을 아랫줄에 작성, 제목 끝에 마침표 금지, 무엇을 했는지 명확하게 작성

szzang-kr commented on Projects/App/Sources/Apply/ViewModel/ApplyViewModel.swift:
```diff
@@ -44,10 +44,14 @@ final class ApplyViewModel: ApplyViewModelProtocol {
-            } onFailure: { [weak self] _ in
-                self?.showToastView.accept(false)
+            .subscribe { [weak self] applyResponse in
+                if applyResponse.code == "S001" {
+                    self?.showToastView.accept((true, applyResponse.message, nil))
+                } else if applyResponse.code == "F009" {
+                    self?.showToastView.accept((false, applyResponse.message, nil))
+                }
```
```suggestion
                if applyResponse.code == "S001" {
                    self?.showToastView.accept((true, applyResponse.message, nil))
                } else  {
                    self?.showToastView.accept((false, applyResponse.message, nil))
                }
```
"F009"가 아닌 에러에대해서도 토스트를 띄울 수 있도록 처리해줘야할 것 같아요!

kimkyunghun3 replied:
수정했습니다!

szzang-kr commented on Projects/App/Sources/Apply/ViewModel/ApplyViewModel.swift:
```diff
@@ -12,15 +12,15 @@ import RxRelay
 
 protocol ApplyViewModelProtocol {
-    var showToastView: PublishRelay<Bool> { get }
+    var showToastView: PublishRelay<(Bool, String?, Error?)> { get }
```
조금 더 깔끔하게 이벤트를 보낼 수 있도록 함께 고민해봅시다. 😭 (cc. @ios-h)

kimkyunghun3 committed:
:recycle: response code 분기 수정

# :tada: = 프로젝트 시작 (#)
# :sparkles: = create 파일 생성 시 (#)
# :bulb: = update 코드 및 UI 업데이트, 기능 구현 시 (#)
# :rocket: = release 배포 시 (#)
# :recycle: = refactor 리팩토링 시 (기능은 같은데 코드가 달라지는경우) (#)
# :memo: = 문서 작성 시 (#)
# :art: = 디자인 시스템 관련 내용 (Color / Asset) (#)
# :hammer: = 테스트 코드 작성 (#)
################

# 커밋 타입 : 커밋 제목 (계정명/저장소명#이슈번호) 형식으로 작성하며 제목은 최대 50글자 정도로만 입력
# 제목을 아랫줄에 작성, 제목 끝에 마침표 금지, 무엇을 했는지 명확하게 작성
